### ISSUE TITLE
[amp-story] Allow players to control system UI controls

### ISSUE THREAD

**Issue by Enriqe** [ampproject/amphtml#30588]
For the amp-story-player, we would like to control the position & visibility of UI elements of the amp-story's system layer (speaker, sharing...). 

Since amp-story knows best where to position them in different scenarios (page-attachment open, tweet expanded, etc...),  it's appropriate for the player to send a message to the story with the required configuration, and the story takes care of the layout logic.

#### Player to story message

`player.sendRequest('customDocumentUI', config);`

#### Config API

(based on https://github.com/ampproject/amphtml/issues/30031)

```
  “controls”: array of objects, each one containing:
    “name”: a string with the name of the button. If the name is part of the default set of buttons (close / (un)mute / share) the following fields are optional but overridable.
    “icon”:  (required when not part of the default set of buttons) accepts  URLS and `data:image` URIs like base64 images and svgs. 
    “event”: (required when not part of the default set of buttons) custom event that will be dispatched when clicked.
    “visibility”: “hidden” or “visible”
    “position”: “end” or “start” string values

```

Example:
```
    {
        "controls": [ 
          {
            "name": "mute", // If name is part of default set: the default event/behavior and icon will be used, but can be overridden. 
            "icon": "icon.jpg",  // Supports URLS and `data:image` URIs like base64 images and svgs.
          },
          {
            "name": "close",
            "position": "start", // Values: "start" or "end".
          },
          {
            "name": "customButton", // If not part of default set, the following fields will be required.
            "icon": "data:image/svg+xml;charset=utf-8,<svg ... ", 
            "event": "customEvent", // Event that will be dispatched when clicked.
          }
        ]
    }
```

#### Listening to Custom Events

Players could then subscribe to events coming from the buttons using the existing viewer message `documentStateUpdate` and checking if it is a player UI event.

```
playerMessaging.registerHandler('documentStateUpdate', (event, data) => {
   if (data.state === 'PLAYER_UI_EVENT`) {
       console.log(data.value); // `amp-story-player-close` (Close button was clicked!)
   }
});
```

@ampproject/wg-stories

**Comment by newmuis** [comment]
nit: should we do the listening for `documentStateUpdate` and re-trigger a javascript event?  That way the only code necessary is standard event listening (either vanilla JS or from the publisher's framework of choice)

If the `documentStateUpdate` event doesn't match one specifies by the UI Controls API, we can just continue to propagate it.

**Comment by Enriqe** [comment]
Yep, that's what I had in mind for the `amp-story-player`. I didn't mention it in this issue because I think @gmajoulet wanted this to be something that any publisher could reference to and implement in their own viewer/players.

**Comment by Enriqe** [comment]
I think we should render at least the footer in the story. There are plans to move some controls (e.g. share) to the footer, and having some elements in the lower part of the story (video loading spinner, sound equalizer), it would make sense for the story to manage these, same as with the header controls.

For the logo, I think we should also render it at the story level since the story would know best where to position it given the other controls' position. It also has the publisher's metadata already in the document, so no extra message would be needed. 

WDYT? @gmajoulet 


> Let's decide on one name for the feature, either "button" or "control" or anything you like, and be consistent on all our APIs.

Sounds good. Landed on controls, will update all the APIs.

**Comment by gmajoulet** [comment]
That all sounds good!

We can make the footer decision later once we know more about our plans but it's great that this API will let us pass its configuration as well, if needed.

**Comment by Enriqe** [comment]
Closed by #30502
